### ISSUE TITLE
NVDA in Chrome does not read out dynamic content of iframe on initial load

### ISSUE THREAD

**Issue by CryptoKiddies** [nvaccess/nvda#11932]
In the latest versions of Chrome (v87+), NVDA is not reading out the content in a dynamically loaded iframe unless focus is moved to the iframe container first. The content is then readable upon tabbing through interactive elements. Another peculiar workaround is to press the Enter key after load, upon which content is read as expected. There does not seem to be anything wrong with the page HTML. This was not an issue in older versions of Chrome.

### Steps to reproduce:

After postback in a pop-up, NVDA doesn't seem to detect the new IFRAME content. If the user will tab-focus outside the iframe (e.g., by pressing Shift+Tab to focus the pop-up's close button) and then tab-focus back into the frame, at that point NVDA will detect the new content and will read out the contents correctly. Pressing the Enter key after load also addresses the issue. I am not certain if this is also an issue in dynamic content fetched from a non-popup context. This only happens on first iframe load. If popup is closed and then steps are recreated, this is no longer an issue, unless the navigation location is reloaded again.

### Actual behavior:

Silent on dynamic iframe content load inside a popup container.

### Expected behavior:

Read out all tabable elements.

#### NVDA installed/portable/running from source:
Installed

#### Does the issue still occur after restarting your computer? 
Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If addons are disabled, is your problem still occuring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by pbsinnett** [comment]
Confirmed. But this is an issue with the actual Chromium engine that I don't even know where or how to begin to report. This bug is present in the new Edge and Electron apps too. Just try using Facebook Messenger in the browser. When you control+tab to another page and receive a message, the browser will not even show the updated content when you go back to the Messenger tab. Also, certain sliders show 50% no matter what their status may be. The best example would probably be the sliders on Twitter videos.

At first I didn't know where it was coming from. I've been experiencing it for a while. After I got an Edge update, I realized it was in the Chromium engine itself. All this to say that this issue can probably be closed and reported in the Chromium issue tracker.

**Comment by GeeeCoin** [comment]
Thanks for confirming. If it includes sliders, seems to be an issue with encapsulated elements, whether they're iframe or shadow dom.

**Comment by Adriani90** [comment]
@CryptoKiddies, @pbsinnett are you still seeing this issue with NVDA 2023.1 and Chrome or Edge 112 or later?

**Comment by pbsinnett** [comment]
Not really. I still run across the very rare occasion where I have to refresh the document content, but I can't reproduce it consistently.

**Comment by Adriani90** [comment]
Thanks for the update. I am closing this as works for me, but in case you find a way to reproduce it reliably please comment on this and we can reopen and continue investigating. or open a new issue with updated steps to reproduce.

**Comment by redmagikarp13** [comment]
I have same problem running NVDA 25.5.5 and Chrome 146 when acessing Moodle 4 website of my university, more specific of TinyMCE. In Firefox Nvda can read all fields of text.

**Comment by seanbudd** [comment]
@leonardcoutinho - This issue was closed 3 years ago. Please consider opening a [new issue](https://github.com/nvaccess/nvda/issues/new/choose) instead of leaving a comment here.

**Comment by redmagikarp13** [comment]
@seanbudd I did too https://github.com/nvaccess/nvda/issues/19897
